chromium_src/content/browser/speech/network_speech_recognition_engine_impl.cc

```diff
@@ -5,46 +5,74 @@
 
 #include <string>
 
-#include "base/feature_list.h"
-#include "base/metrics/field_trial_params.h"
+#include "base/containers/flat_map.h"
+#include "brave/components/brave_service_keys/brave_service_key_utils.h"
 #include "brave/components/constants/brave_services_key.h"
+#include "brave/components/speech_to_text/buildflags.h"
+#include "brave/components/speech_to_text/features.h"
+#include "services/network/public/cpp/resource_request.h"
 #include "url/gurl.h"
 
-namespace content {
-namespace google_apis {
+namespace content::google_apis {
 std::string GetAPIKey() {
   return BUILDFLAG(BRAVE_SERVICES_KEY);
 }
-}  // namespace google_apis
-}  // namespace content
+}  // namespace content::google_apis
 
 namespace {
 
-BASE_FEATURE(kSttFeature, "speech_to_text", base::FEATURE_DISABLED_BY_DEFAULT);
-const base::FeatureParam<std::string> kSttUrl{&kSttFeature, "web_service_url",
-                                              ""};
+std::string GetWebServiceBaseUrl(const char* web_service_base_url_for_tests) {
+  if (base::FeatureList::IsEnabled(stt::kSttFeature)) {
+    return stt::kSttUrl.Get();
+  }
+  // Fallback to for-tests URL.
+  return web_service_base_url_for_tests;
+}
+
+void AddBraveHeaders(network::ResourceRequest* request,
+                     const std::string& request_key) {
+  DCHECK(request && !request->method.empty() && request->url.is_valid());
+
+  const std::string brave_sticky_session_cookie =
+      "Brave-stt-sticky=" + request_key;
+  request->headers.SetHeader(net::HttpRequestHeaders::kCookie,
+                             brave_sticky_session_cookie);
 
-std::string GetWebServiceBaseUrl(const char* web_service_base_url) {
-  if (base::FeatureList::IsEnabled(kSttFeature)) {
-    return kSttUrl.Get();
+  constexpr const char kRequestKey[] = "request-key";
+  request->headers.SetHeader(kRequestKey, request_key);
+
+  const base::flat_map<std::string, std::string> headers{
```

Comment:
since we are unable to sign over the body ( via the digest header ) due to streaming, there is nothing that prevents someone from re-using the same final authorization value ( and `request-key` ) to process multiple separate audio streams.

I would recommend that we sign over the `Date` header and ensure that server side we reject requests that are too far from the current time. assuming we have a max session timeout, that could be used for the tolerance value. ( e.g. if the max session is 2m, then we could accept requests whose `Date` header is within +/- 2m from the current time. )